### PR TITLE
Adding cucushift to upgrade prow job

### PR DESCRIPTION
**Pull request by jhou1**
WIP, this attempts to run cucushift upgrade-prepare, upgrade-check tests in Prow.

cc @liangxia @JianLi-RH @dis016 @pruan-rht

### CODE REVIEW

**Review by dis016 — COMMENTED** [review]
/lgtm

File: ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# run upgrade
+function upgrade() {
+    cmd="oc adm upgrade --to-image=${TARGET_RELEASE} --force=true --allow-explicit-upgrade=true --allow-upgrade-with-warnings=true"
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh:9]: Is this `--allow-upgrade-with-warnings` for some test purpose?

File: ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    echo "Running command: ${cmd}"
+    eval "${cmd}"
+}
+
+function monitor_cvo() {
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh:14]: `monitor_clusterversion` seems more suitable.

File: ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    INTERVAL=600
+    CNT=12
+
+    while [ $((CNT)) -gt 0 ]; do
+        # or history
+        version_full=$(oc get clusterversion/version -o json)
+        image=$(echo "${version_full}" | jq -r '.status.history[0].image')
+        if [[ "${image}" != "${TARGET_RELEASE}" ]]; then
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh:22]: This check is not enough. When upgrade is triggered, the image is updated. We need check state and version together.

File: ci-operator/step-registry/cucushift/aws/ipi/upgrade/cucushift-aws-ipi-upgrade-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+workflow:
+  as: cucushift-aws-ipi-upgrade
+  steps:
+    pre:
+      - chain: ipi-aws-pre
+      - ref: cucushift-pre
+      - ref: cucushift-upgrade-pre
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/aws/ipi/upgrade/cucushift-aws-ipi-upgrade-workflow.yaml:7]: Hmm..I still have several questions about the steps here. Maybe we can have a sync later.

**Comment by jhou1** [comment]
Per the sync with @jiajliu , we will hold this and have a discussion about integrating current upgrade, upgrade-healthcheck and cucushift e2e together into this job.
/hold

**Pull request merged**